base/file.jl

```diff
@@ -500,29 +500,30 @@ function mktemp(parent::AbstractString=tempdir(); cleanup::Bool=true)
     return (filename, Base.open(filename, "r+"))
 end
 
-function tempname()
-    parent = tempdir()
+function tempname(parent::AbstractString=tempdir(); cleanup::Bool=true)
     seed::UInt32 = rand(UInt32)
     while true
         if (seed & typemax(UInt16)) == 0
             seed += 1
         end
         filename = _win_tempname(parent, seed)
         if !ispath(filename)
```

Comment:
This is a race condition,
do we want to make it threadsafe?
Wouldn't entirely make it safe since external processes also could create clashing names.